File: source-code/LeilaSalaoFrontEnd/src/app/features/scheduling/dto/create.scheduling.request.dto.ts
export interface CreateSchedulingRequestDto {
    dateHours: string
    observations: string | null | undefined
    origin: SchedulingOrigin | null
    servicesIds: string[]
}

export interface SchedulingServiceDto {
    id: string
    name: string
}

export type SchedulingStatus =
    | 'SCHEDULED'
    | 'DONE'
    | 'CANCELLED'

export type SchedulingOrigin =
    | 'APP'
    | 'PHONE_NUMBER'

File: source-code/LeilaSalaoApi/src/main/java/com/cabeleleira/leila/salao/dto/CreateSchedulingRequestDTO.java
package com.cabeleleira.leila.salao.dto;

import com.cabeleleira.leila.salao.enums.SchedulingOrigin;
import jakarta.validation.constraints.Future;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

public record CreateSchedulingRequestDTO(
        @NotNull(message = "A data e hora do agendamento é obrigatória")
        @Future(message = "A data do agendamento deve ser no futuro")
        LocalDateTime dateHours,

        @Size(max = 500, message = "As observações devem ter no máximo 500 caracteres")
        String observations,

        SchedulingOrigin origin,

        @NotEmpty(message = "É necessário informar pelo menos um serviço")
        List<UUID> servicesIds
) { }
